jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -141,6 +147,36 @@ def _parse_spectral_cube(app, file_obj, data_type='flux', data_label=None):
     # TODO: SpectralCube does not store data quality information
 
 
+def _parse_spectrum1d_3d(app, file_obj):
+    # Load spectrum1d as a cube
+
+    for attr in ["flux", "mask", "uncertainty"]:
+        if attr == "mask":
+            flux = getattr(file_obj, attr) * file_obj.flux.unit
+        elif attr == "uncertainty":
+            flux = getattr(file_obj, attr)
+            flux = u.Quantity(flux.array) * file_obj.flux.unit
```

Comment:
I think this should be `u.Quantity(flux.array, file_obj.flux.unit)`. If `flux.array` already has unit, you are multiplying `file_obj.flux.unit` to it instead of recasting.

Reply:
If performance is a problem, also consider whether you can get away with `copy=False` or `subok=True`. See https://docs.astropy.org/en/latest/api/astropy.units.Quantity.html#astropy.units.Quantity